integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JMSInboundHttpTenantTestCase.java

```diff
@@ -0,0 +1,166 @@
+        inboundAdminClient2.addInboundEndpoint(createJMSInboundEndpoint(TENANT2_INBOUND_EP, TENANT2_QUEUE).toString());
+        Assert.assertTrue(Utils.checkForLog(logViewerClient, TENANT1_SYMBOL, 10),
+                          "Message is not received by tenant: " + TENANT1);
+        Assert.assertTrue(Utils.checkForLog(logViewerClient, TENANT2_SYMBOL, 10),
+                          "Message is not received by tenant: " + TENANT2);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        super.cleanup();
+        inboundAdminClient1.removeInboundEndpoint(TENANT1_INBOUND_EP);
+        inboundAdminClient2.removeInboundEndpoint(TENANT2_INBOUND_EP);
+    }
+
+    /**
+     * Creates the synapse configuration for an inbound endpoint given the inbound endpoint name and the queue name
+     * to be connected to.
+     *
+     * @param inboundEPName the name of the inbound endpoint ot be created
+     * @param queueName     the name of the queue to be connected to
+     * @return the synapse configuration of the inbound endpoint
+     * @throws Exception if an error occurs during the conversion of the string representation of the inbound
+     *                   endpoint to an OMElement
+     */
+    private OMElement createJMSInboundEndpoint(String inboundEPName, String queueName) throws Exception {
+        OMElement synapseConfig;
+        synapseConfig = AXIOMUtil.stringToOM(createInboundEndpointString(inboundEPName, queueName));
+        return synapseConfig;
+    }
+
+    /**
+     * Creates a soap message with a symbol specified.
+     *
+     * @param symbol the symbol to be added
+     * @return the soap message
+     */
+    private String createMessage(String symbol) {
+        return "<?xml version='1.0' encoding='UTF-8'?>"
+               + "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\"\n"
+               + "                   xmlns:ns= \"http://services.samples\"\n"
+               + "                   xmlns:xsd=\"http://services.samples/xsd\">\n"
+               + "      <soapenv:Header/>\n"
+               + "       <soapenv:Body>\n"
+               + "           <ns:getQuote >\n"
+               + "               <ns:request>\n"
+               + "                   <ns:symbol>" + symbol + "</ns:symbol>\n"
+               + "               </ns:request>\n"
+               + "           </ns:getQuote>\n"
+               + "       </soapenv:Body>\n"
+               + "</soapenv:Envelope>";
+    }
+
+    /**
+     * Creates a JMS inbound endpoint provied the name of the inbound endpoint and the name of the queue to be
+     * listening to.
+     *
+     * @param inboundEPName the inbound endpoint name
+     * @param queueName     the name of the queue to be connected to
+     * @return the string representation of inbound endpoint configuration
+     */
+    private String createInboundEndpointString(String inboundEPName, String queueName) {
```

Comment:
Why not import these configurations from a file instead of string literal ?

Reply:
That is because the inbound endpoint name and the queue name could change.

Reply:
That is because the enbound endpooint name and the queue name could change in this test case. We'll need to have multiple config files to cater that.